event properties object instruction (#261)
amplitude/amplitude-dev-center: docs/data/sources/google-tag-manager-client.md

| --- | --- |
| `Event Type` | `string`. The name of the event. |

##### Event Properties Basic

| Name  | Description |
| --- | --- |
| `Property Name` | `string`. The name of the event property. |
| `Property Value` | `string`. The value of the event property. |

##### Event Properties Object

> Choose the object format for the GTM variable to define event properties. It's important to note that if there are any duplicate keys, the event properties object will overwrite the basic event properties. Any input not in object format will be ignored.

Reviewer: ```suggestion
Choose the object format for the GTM variable to define event properties. Use unique values for keys in the object. Amplitude overwrites any basic event properties that have duplicate keys. Amplitude ignores any inputs not in the object format.
```

Author: Thanks for reviewing @markzegarelli .
Basically, there are 2 event properties. **EVENT PROPERTIES BASIC** and **EVENT PROPERTIES OBJECT**. I What I want to clarify here is if there is a duplicate key between **EVENT PROPERTIES BASIC** and **EVENT PROPERTIES OBJECT**, the value in **EVENT PROPERTIES OBJECT** will overwrite the value in **EVENT PROPERTIES BASIC**.  

I feel here is a misunderstanding.

Author: Thanks @markzegarelli.
The feature has not been published yet, I will merge this PR after the feature release.

| --- | --- |
| `Event Type` | `string`. The name of the event. |

> ##### Event Properties Basic

Reviewer 2: Should we call this Event properties key value pairs?

Author: I think both ways are setting the key-value pair. But the difference is it's by individual or bulk set. 
I use individual operation vs Bulk set operation for the identify properties. How do you like it?
May be here we use  **Individual Event Properties**?

Reviewer 2: Yea maybe lets go with individual event properties.

| Name  | Description |
| --- | --- |
| `Property Name` | `string`. The name of the event property. |
| `Property Value` | `string`. The value of the event property. |

##### Event Properties Object

> Select a GTM variable that returns a valid event properties object. This will overwrite the **Event Properties Basic** event properties if there has any duplicate key. Amplitude ignores any inputs not in the object format.

Reviewer: ```suggestion
Select a GTM variable that returns a valid event properties object. This overwrites the **Event Properties Basic** event properties if there are any duplicate keys. Amplitude ignores any inputs not in the object format.
```

> Select a GTM variable that returns a valid event properties object. This overwrites the **Event Properties Basic** event properties if there are any duplicate keys. Amplitude ignores any inputs not in the object format and any value under `user_properties` key.

Reviewer 2: Can we call out the Google variable that has both event properties and user properties in it? And provide an example of what that looks like and what will be ignored in that object.

Reviewer 2: And actually, we should just advice that they use that variable always to keep it simple.

Author: Added another shared section to introduce **Google Tag: Event Settings** and added it to both the user properties and event properties section.

| --- | --- |
| `Event Type` | `string`. The name of the event. |

##### Individual Event Properties

| Name  | Description |
| --- | --- |
| `Property Name` | `string`. The name of the event property. |
| `Property Value` | `string`. The value of the event property. |

##### Event Properties Object

> Select a GTM variable that returns a valid event properties object. This overwrites the **Event Properties Basic** event properties if there are any duplicate keys. Amplitude ignores any inputs not in the object format and any value under `user_properties` key.

Reviewer 2: ```suggestion
Select a GTM variable that returns a valid event properties object. This overwrites the **Individual event properties** if there are any duplicate keys. Amplitude ignores any inputs not in the object format and any value under `user_properties` key.
```

Author: fixed.
